### PR TITLE
Env Variable TEST_MODE works as expected

### LINKED ISSUE #906: RPF: Configure DVR to re-record existing tests
DVR concurrently looks to see if a recording exists, and if it does, it will consider itself in playback mode. We would prefer to explicitly call out whether we want DVR to record.

Modes:
- record: record tests or, if they already exist, re-record them
- playback: playback tests if they exist; if not, fail

This should honor the TEST_MODE environment variable we are setting.

### PR DESCRIPTION
**Pull request by jairmyree**
Enforce that DVR record-playback honors environment variables
Live will ignore recordings and always run live
Record will ignore recordings, run live, and create a new recording
Playback will only play recordings and fail if the recording isn't valid or does not exist.
Closes #906

### CODE REVIEW

**Review by tjprescott — COMMENTED** [review]
This looks good. I'm assuming it depends on a yet-to-be-opened PR against DVR?

**Comment by tjprescott** [comment]
Closes #906.

**Comment by tjprescott** [comment]
Depends on: https://github.com/tjprescott/DVR/pull/1

**Review by jairmyree — COMMENTED** [review]
@tjprescott Please take a look when you get a chance.

File: sdk/test/AzureTest/DVRSessionTransport.swift [hunk 1]
```diff
@@ -64,8 +64,14 @@ public class DVRSessionTransport: TransportStage {
         }
         session = Session(outputDirectory: outputDirectory, cassetteName: cassetteName)
         if environmentVariable(forKey: "TEST_MODE", default: "playback") == "record" {
+            session?.recordMode = .newEpisodes
```
Comment (tjprescott) [inline sdk/test/AzureTest/DVRSessionTransport.swift:67]: If in record mode, this should be `.all`. I don't think we'll use `.once` or `.newEpisodes` at all.

**Review by tjprescott — CHANGES_REQUESTED** [review]

**Review by tjprescott — APPROVED** [review]

**Pull request merged**